### DISCUSSION TITLE
Need help to add custom filtering in Debug Console

### DISCUSSION THREAD

**Discussion by eli88p** [eclipse-theia/theia#8509] [General]
Hello,
I need to add more filtering options in Debug Console
can i do it?

Thanks in advance
Eli.

**Comment by vince-fugnitto** [comment]
@eli88p thank you for the discussion, do you mind providing additional details? (I assume you are referring to the debug console from the framework itself contributed by `@theia/console`?) What type of filtering to you require for your use-case?

**Reply by eli88p** [reply]
@vince-fugnitto thanks for your quick response...
i want to do something like in the picture and i need advice, what the best way to do it
![cap1](https://user-images.githubusercontent.com/11608934/93341882-53245080-f837-11ea-8efa-90c50f82c5b8.png)

**Reply by vince-fugnitto** [reply]
@eli88p you can inspire yourself from the implementation of the `severity` filter (all, errors, warnings, info, logs), in order to implement your secondary filter. It would be the best reference for your own custom implemntation:

- https://github.com/eclipse-theia/theia/commit/f2b3556935a3aa9d277481e622d3025c07d6ed17

If you want to add your custom filter, you will need to extend the base class and rebind your component through `inversify` (dependency injection) given that the filter cannot be upstreamed and will need to be maintained by yourself.
